conformance/lang/values-types-and-variables/map_types.balt

```diff
@@ -0,0 +1,211 @@
+}
+
+Test-case: output
+Description: Test mapping type-descriptor being covariant in the types of their members.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, byte-type-descriptor, 
+        anydata-type-descriptor, mapping-constructor-expr, int-literal
+
+public function main() {
+    map<byte> a = {a: 1, b: 2};
+    testMap(a);
+
+    map<int> b = a;
+    testMap(b);
+}
+
+function testMap(map<anydata> m) {
+    io:println(m); // @output {"a":1,"b":2}
+                   // @output {"a":1,"b":2} 
+}
+
+Test-case: output
+Description: Test mapping type-descriptor being covariant in the types of their members, at runtime.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, byte-type-descriptor, 
+        anydata-type-descriptor, mapping-constructor-expr, int-literal, string-literal, is-expr
+
+public function main() {
+    map<byte> a = {a: 1, b: 2};
+    testMap(a);
+
+    map<int> b = a;
+    testMap(b);
+
+    map<int> c = {x: 1, y: 2, z: 3};
+    testMap(c);
+
+    map<anydata> d = {a: 1, b: "hello world"};
+    testMap(d);
+
+    testMap({a: 1, b: 2});
+}
+
+function testMap(map<anydata> m) {
+    io:println(m is map<int>); // @output true
+                               // @output true
+                               // @output true
+                               // @output false
+                               // @output false
+}
+
+Test-case: error
+Description: Test mapping type-descriptor being covariant in the types of their members, via incompatible assignments.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, byte-type-descriptor, 
+        string-type-descriptor, mapping-constructor-expr
+
+public function main() {
+    map<int> a = {};
+    testMap(a); // @error map of int is not a subtype of map of byte
+
+    map<string> b = a; // @error map of int is not a subtype of map of string
+}
+
+function testMap(map<byte> m) {
+}
+
+Test-case: panic
+Description: Test inherent type violation of a map.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, anydata-type-descriptor, 
+        mapping-constructor-expr, int-literal, string-literal, assignment-stmt, member-access-lvexpr, lvexpr
+
+public function main() {
+    map<int> a = {a: 1, b: 2};
+    testMap(a, "hello");
+}
+
+function testMap(map<anydata> m, anydata value) {
+    m["k"] = value; // @panic cannot insert a string value as a field of a map<int>
+}
+
+Test-case: panic
+Description: Test inherent type violation of a map with a value that looks like the constraint.
+Labels: map-type-descriptor, mapping-type-descriptor, int-type-descriptor, anydata-type-descriptor, 
+        byte-type-descriptor, mapping-constructor-expr, int-literal, string-literal, assignment-stmt, 
+        member-access-lvexpr, lvexpr
+
+public function main() {
+    map<map<byte>> a = {x: {a: 1, b: 2}};
```

Comment:
I would use `byte` only if you want to test specifically the functionality of `byte`. For subtype relationships, it's simpler to use

int <: int? <: any

Reply:
Updated the `byte`, `int`, and `anydata` usage with `int`, `int?`, and `any`.